tools/scripts/run_protocol_tests.py

```diff
@@ -0,0 +1,136 @@
+﻿#!/usr/bin/env python3
+
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0.
+
+"""
+A script to run protocol tests for AWS SDK for C++:
+Provided --build_dir, it runs each built protocol test found.
+For each test executed, a mock http server is launched in parallel.
+"""
+import argparse
+import os
+import shutil
+import signal
+import subprocess
+import time
+
+SCRIPTS_DIR = os.path.dirname(__file__)
+PROTO_TEST_MOCK_HANDLER = f"{SCRIPTS_DIR}/protocol_tests_mock_server/protocol_tests_mock_server.py"
+
+
+def parse_arguments() -> dict:
+    """Parse arguments and/or print help
+
+    :return: dict with parsed arguments
+    """
+    parser = argparse.ArgumentParser(description="Protocol tests driver for AWS-SDK-CPP")
+    parser.add_argument("--debug", action="store_true", help="Print additional information")
+    parser.add_argument("--build_dir", action="store", required=True, help="Path to a directory with built SDK")
+    args = vars(parser.parse_args())
+    arg_map = {"debug": args.get("debug", False)}
+    arg_map["build_dir"] = args.get("build_dir")
+    return arg_map
+
+
+class MockHttpServerHandle:
+    """A RAII-style helper wrapper around mini mock http server.
+    Launches a server on construction and kills it on destruction
+    """
+
+    def __init__(self):
+        python_cmd = [shutil.which("python3"), PROTO_TEST_MOCK_HANDLER]
+        self.process = subprocess.Popen(python_cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        print(f"Started mock server with PID {self.process.pid}")
+
+    def __del__(self):
+        """Destructor that sends Ctrl-C/SIGINT to the subprocess."""
+        if self.process and self.process.poll() is None:
+            print(f"Sending SIGINT to subprocess with PID {self.process.pid}")
+            if os.name == 'nt':  # Windows
+                self.process.send_signal(signal.CTRL_C_EVENT)
+            else:  # Unix-like
+                self.process.send_signal(signal.SIGINT)
+
+            time.sleep(0.5)
```

Comment:
do we need static sleep? what are we waiting for?

Reply:
we need to wait a bit to let the process and OS to terminate the process.